### PR TITLE
Typing

### PR DESCRIPTION
**Pull request by StanHannebelle**
Hello!

Here is a migration to typescript for sls-test-tools.
github.com/airbnb/ts-migrate helped me to implement this migration.

I will improve typing (removing any types...) in a coming PR but this one is a good first step to migrate smoothly to ts :)

### CODE REVIEW

**Review by LukeYianni — CHANGES_REQUESTED** [review]
Looks mostly good to me! Could you change some of the _any_ types to one more specific? Thinking particularly in src/helpers/eventBridge.ts

**Comment by StanHannebelle** [comment]
Thanks Luke! I just pushed a fix to remove most "any" types

File: src/assertions/toHaveContentTypeEqualTo/index.ts
Comment (CorentinDoue) [inline src/assertions/toHaveContentTypeEqualTo/index.ts:25]:
```suggestion
        object.ContentType ?? "undefined"
```

File: src/assertions/utils/globalTypeChecker.ts
Comment (CorentinDoue) [inline src/assertions/utils/globalTypeChecker.ts:4]: You can let `any` for this one because it's little complicated and unecessary to type expect.

But for the next time you want to write a typeguard which test a attribute existence I recommand you this one
```ts
export const hasAttributes = <T extends string>(
    element: unknown,
    attributes: T[]
): element is Record<T, unknown> => {
    if (element === undefined || element === null) {
        return false;
    }
    return attributes.every((attribute) =>
        Object.prototype.hasOwnProperty.call(element, attribute)
    );
};
```
from [the fresh article of Fatine about typeguards](https://blog.theodo.com/2022/01/typescript-replace-as-typeguards/)

File: src/helpers/utils/removeUndefinedMessages.ts
Comment (CorentinDoue) [inline src/helpers/utils/removeUndefinedMessages.ts:23]:
```suggestion
    (messageHandler): messageHandler is {
        Id: string;
        ReceiptHandle: string;
      } =>
      messageHandler.Id !== undefined &&
      messageHandler.ReceiptHandle !== undefined
  );
```
to avoid cast.
And you can put
```
{
        Id: string;
        ReceiptHandle: string;
      }
```
in a named type to avoid duplication with the return type of the function

**Review by CorentinDoue — COMMENTED** [review]
Good job

**Review by CorentinDoue — CHANGES_REQUESTED** [review]
Don't forget to add the type of the matchers in a jest namespace as suggested by Valentin. 

Example of lib which declares new matchers and type them: https://github.com/jest-community/jest-extended/blob/main/types/index.d.ts

**Comment by lukey-aleios** [comment]
This is looking great! Absolutely smashing it 💪

LGTM so I'll leave it to @CorentinDoue to approve once the comments he made are resolved

**Comment by StanHannebelle** [comment]
Thanks @lukey-aleios and @CorentinDoue !
I corrected my code after your review Corentin and added the type of the jest matchers

**Review by CorentinDoue — APPROVED** [review]

**Comment by CorentinDoue** [comment]
It looks good to me :) 

@lukey-aleios I think you can merge.

We are planning to continue to contribute to `sls-test-tools`. It could be easier if you add write rights to @valentinbeggi @StanHannebelle and me on the project. We promise we will be good contributors ;)

**Comment by lukey-aleios** [comment]
Looks great guys! I'll merge it in.

I'd personally be incredibly thankful to have you as long term
contributors, I'll speak to Ben about getting you added.

Luke

**Review by lukey-aleios — APPROVED** [review]

**Comment by smbkr** [comment]
Hi, I'm attempting to use this library with a Typescript project. It currently fails with error TS7016: Could not find a declaration file for module 'sls-test-tools'. I am using version 0.13.8 which was released prior to this PR being merged. Is there a pre-release version I can use with Typescript, or do you plan to make an official release soon?

**Pull request merged**
